### ISSUE TITLE
Copy/cut by tag rather than selection

### ISSUE THREAD

**Issue by hektr** [ranger/ranger#1987]
Ranger run selected file, but I want to run current file. 

When I select file(s) and then, want to open some file, only first selected file is opened. 
I tried to use tags instead of selecting files, but commands like `cut %t` dont work, so, I should use `shell mv %t somedir/` instead.

I think, it's needed a command for making tagged files selected. 
It will make it more easy copy-pasting and deleting tagged files, using ranger commands.
What do you think??

**Comment by mrgoodvin** [comment]
> Ranger run selected file, but I want to run current file.

They can additionally select the desired file. After highlighting the files I need, I move the cursor up to the highlighted one, but this habit remains after other FMs, which I can not get rid of. I have no opinion on this issue, but I am quite happy with this and that.

> When I select file(s) and then, want to open some file, only first selected file is opened.

My ranger opens all the selected files, only selected. If you want open some file you can use `%f`. It does not reset the selection.

> I tried to use tags instead of selecting files, but commands like cut %t dont work, so, I should use shell mv %t somedir/ instead.

> I think, it's needed a command for making tagged files selected.
> It will make it more easy copy-pasting and deleting tagged files, using ranger commands.

_do not quite understand you.
1. Do your use `cut %t` in console, why?
_am against the use of qualifiers in the console. This will complicate the processing of the console, because `%s` may be the crazy file name. It is a good idea to create commands for copying file names and paths to the console (keyboard shortcuts -> C-keys for the console).
Yes, now file name qualifiers are used in the console.
When I say console, I mean the console visible to the user.

2. If you use in rc.conf...
Problem with `%s` `%t` `%p`, etc are already registered. We know about this problem.

-

Typically, tags are not used to copy, delete, open etc. They are used for specific operations and creating unified methods seems not logical. Unless it is due to the specifics of the ranger. If you use often tags, you can create methods/commands/scripts for yourself.

imho

**Comment by toonn** [comment]
That's how selection works, it is either the highlighted file or if you have explicitly marked any files it's just the explicitly marked files. This allows for more freedom to look around for files to mark without having to go all the way back to a marked file before running a command.

cut and copy only work on the selection. Making them work on paths (as passed with a macro like `%t`) might be hard because ranger works on file objects, not paths.

You can already mark tagged files using `:mark_tag`.
I agree that some commands should interact with tags in a more streamlined fashion. Especially those commands that ordinarily work on the selection.

**Comment by hektr** [comment]
> You can already mark tagged files using `:mark_tag`.
Yes, it is what I wanted :) Thank you.

**Comment by hektr** [comment]
> I agree that some commands should interact with tags in a more streamlined fashion. Especially those commands that ordinarily work on the selection.
>
I think, mark_tag is enough for this purpose. It is not breaking the idea, that only selected or current file is the object of file operations. It keeps things simple and don't require rewritting code :)

btw, is there any way to copy filename ?

**Comment by toonn** [comment]
You can copy filenames and paths using the `:yank` command. `yn` is bound to copying the name by default, requires `xclip` or `xsel`.

I'm keeping this open as a feature request because imo tags should just be permanent selections.

**Comment by hektr** [comment]
> You can copy filenames and paths using the `:yank` command. `yn` is bound to copying the name by default, requires `xclip` or `xsel`.
> 
thanks! 

> I'm keeping this open as a feature request because imo tags should just be permanent selections.
>
It's not handy. If i want to choose some pics from a folder and move them to another folder, i need to see at pic, then decide whether mark this pic or not, then, do the same for all other pics. So, i need ranger run only current file and not 'marked' ones. Tags solve this problem. You can mark files using tags and make them selected when you need to copy them.

**Comment by mrgoodvin** [comment]
> It's not handy. If i want to choose some pics from a folder and move them to another folder, i need to see at pic, then decide whether mark this pic or not, then, do the same for all other pics. So, i need ranger run only current file and not 'marked' ones. Tags solve this problem. You can mark files using tags and make them selected when you need to copy them.

@hektr
You're right. But this can already be done without writing code. Need comand only on shortcut key. Try to do (`console shell cp -irv %t /to_folder_target`) with files names that do not have a spaces. You should succeed. I will try to add these commands to my configuration. But files with spaces or files with crazy names will not be processed correctly until we solve the problem of qualifiers `%s` and etc.
I just don’t use tags.

If we could open a certain (`%f`) file on a previously made selection, you would not even have the idea of using tags.

**Comment by hektr** [comment]
@mrgoodvin using space in a filename is not good style of naming files 😂 You should escape special sign in file name using backslash before it, or take the whole filename in quotes.

**Comment by mrgoodvin** [comment]
Yes, it true ). Sometimes we already got the names and we didn’t choose them.

**Comment by hektr** [comment]
@mrgoodvin it works, I tried. `shell cp -irv %s ..` copies selected files(filenames have space) in parent directory. So, I don't know what is the problem with %s.

**Comment by mrgoodvin** [comment]
> @mrgoodvin it works, I tried. `shell cp -irv %s ..` copies selected files(filenames have space) in parent directory. So, I don't know what is the problem with %s.

Try then with the following file names: `"11'' "' '"'`, `"11'' "' '"2`

**Comment by hektr** [comment]
@mrgoodvin that is not real file names, but i've tried this and it works :) May be, something is wrong with your locale configuration or something like that

**Comment by mrgoodvin** [comment]
Hmm ... Yes, I thought so too. It is quite possible, but there is also the processing logic that I observe in FM. Show what your ranger console looks like after such a command.
How after console processing distinguishes file names?

**Comment by mrgoodvin** [comment]
Let's assume that in the system console I want to perform some actions on file names that are not on the list. For example, they should be created.
I want to introduce them, so I use the ranger console. How does a ranger determine that I am a home if I use a name that is not clear to him? You probably all do it in the shell. While I'm talking about console + shell.


Ok, let me describe in more detail for my settings. For we can talk about it for a long time.
Sorry, I will be busy. Actually, even now I have little time.
Thank you for your interest.

And we should not talk about it here :). Sorry guys.

If it’s not difficult for you, can you do one thing again for this file name `"file '1' "name "" "1'`?

map fra eval fm.open_console("rename %f", position=7)
What you see in console?
keys -> fra

**Comment by toonn** [comment]
@hektr, I think you misunderstand me. My point is that any commands that act on the selection *should* allow you to pass a tag instead. Because tags are basically permanent selections. For example `cut *` would cut all the files tagged with the default tag.

The issue mrgoodvin is talking about only happens when you combine `:shell` with `:console`. `:shell %s` -> proper escaping of filenames in the selection, `:console shell %s` -> no escaping of filenames in the selection.

**Comment by hektr** [comment]
@toonn yes, I've misunderstood you. Yes, it's really good idea!

**Comment by toonn** [comment]
You could create a custom command or just a keybinding that does that.
What I have in mind wouldn't mess with a user's selection.

**Comment by hektr** [comment]
Thanks! I understood
